beacon/blockchain/payload_test.go

```diff
@@ -100,46 +101,36 @@ func TestOptimisticBlockBuildingRejectedBlockStateChecks(t *testing.T) {
 
 	// register async call to block building
 	var wg sync.WaitGroup          // useful to make test wait on async checks
-	var ch = make(chan struct{})   // useful to serialize build block goroutine and avoid data races
 	stateRoot := st.HashTreeRoot() // track state root before the changes done by optimistic build
 	latestHeader, err := st.GetLatestBlockHeader()
 	require.NoError(t, err)
 	latestHeader.SetStateRoot(st.HashTreeRoot())
 	expectedParentBlockRoot := latestHeader.HashTreeRoot()
 
-	b.EXPECT().RequestPayloadAsync(
-		mock.Anything, mock.Anything, mock.Anything,
-		mock.Anything, mock.Anything, mock.Anything, mock.Anything,
-	).Run(
-		func(
-			_ context.Context,
-			st *state.StateDB,
-			slot, timestamp math.U64,
-			parentBlockRoot common.Root,
-			headEth1BlockHash, finalEth1BlockHash common.ExecutionHash,
-		) {
+	b.EXPECT().RequestPayloadAsync(mock.Anything, mock.Anything).Run(
+		func(_ context.Context, r *builder.RequestPayloadData) {
 			defer wg.Done()
-			<-ch // wait for block verification to finish. This avoids data races over state reads
 			genesisHeader := genesisData.ExecutionPayloadHeader
 			genesisBlkHeader := core.GenesisBlockHeader(cs.GenesisForkVersion())
 			genesisBlkHeader.SetStateRoot(stateRoot)
 
-			require.Equal(t, timestamp, consensusTime+1)
+			require.Equal(t, consensusTime+1, r.Timestamp)
 
-			require.Equal(t, genesisHeader.GetBlockHash(), headEth1BlockHash)
+			require.Equal(t, genesisHeader.GetBlockHash(), r.HeadEth1BlockHash)
 
-			require.Equal(t, expectedParentBlockRoot, parentBlockRoot)
+			require.Equal(t, expectedParentBlockRoot, r.ParentBlockRoot)
 
-			require.Empty(t, finalEth1BlockHash)          // this is first block post genesis
-			require.Equal(t, constants.GenesisSlot, slot) // genesis slot in state
-			var stateSlot math.Slot
-			stateSlot, err = st.GetSlot()
-			require.NoError(t, err)
-			require.Equal(t, constants.GenesisSlot, stateSlot)
+			require.Empty(t, r.FinalEth1BlockHash)            // this is first block post genesis
+			require.Equal(t, constants.GenesisSlot+1, r.Slot) // rebuild block on top of genesis
```

Comment:
This is actually a change wrt to `main` but I believe `main` is wrong.
The scenario is:
- a block `B_0` of height `h` is already final (so in `state`)
- a block `B_1` of height `h+1` is being verified and it is rejected.

What should be the height of the next block being built optimistically?
We should build a sibling of `B+1` so the requested height should be `B_0` height +1.
This is what it is asserted in this test (which does not happen in prod)